### DISCUSSION TITLE
Can't install Homebrew on linux

### DISCUSSION THREAD

**Discussion by MatteoGauthier** [Homebrew/discussions#5601] [Linux] [answered]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
not applicable
```

### Output of `brew doctor`

```shell
not applicable
```

### Description of issue

I'm trying to install homebrew, using the command provided on the home page. Here is the output
```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
```
==> Checking for `sudo` access (which may request your password)...
Don't run this as root!
```

And it exit with an 1 error code.

**Comment by SMillerDev** [comment]
Did you run the command as root?

**Reply by MatteoGauthier** [reply]
Yes my i'm using the `root` which is the default on my VPS.
```
id
uid=0(root) gid=0(root) groups=0(root)
```

Is there a way to maybe create an user non-root just for this installation ?

**Reply by carlocab** [reply, marked as answer]
> Is there a way to maybe create an user non-root just for this installation ?

Yes, probably better to create a `linuxbrew` user with something like
```
sudo useradd --user-group --create-home linuxbrew
```
and then run the install script as said `linuxbrew` user. You may want to give the `linuxbrew` password-less `sudo` access with something like
```
sudo tee /etc/sudoers.d/linuxbrew <<<"linuxbrew ALL=(ALL) NOPASSWD:ALL"
```

**Reply by MatteoGauthier** [reply]
Thanks a lot for your explanation ! Helped me a lot
